### PR TITLE
Cleanup and address obsidian releases feedback

### PR DESCRIPTION
**Pull request by kristenbrann**
https://github.com/obsidianmd/obsidian-releases/pull/1752#issuecomment-1469805304

### CODE REVIEW

File: src/UserSettings.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { App, PluginSettingTab, Setting, SliderComponent } from 'obsidian'
```
Comment (kristenbrann) [inline src/UserSettings.ts:1]: unused import

File: src/VectorStore.ts [hunk 2]
```diff
@@ -79,8 +79,8 @@ export class VectorStore {
 		return new Map(result)
 	}
 
-	getByFilePath(filePath: string): Vector {
-		return this.filePathToVector.get(filePath) as Vector
+	getByFilePath(filePath: string): Vector | undefined {
+		return this.filePathToVector.get(filePath)
```
Comment (kristenbrann) [inline src/VectorStore.ts:83]: no need to cast here

File: src/appContext.ts
Comment (kristenbrann) [inline src/appContext.ts:1]: unused code from initially incorporating react

File: src/main.ts [hunk 3]
```diff
@@ -143,8 +143,12 @@ export default class VaultChat extends Plugin {
 		const searchResults = Array.from(nearestVectors.keys())
 		const hydratedResults = []
 		for (const searchResult of searchResults) {
-			const abstractFile = app?.vault.getAbstractFileByPath(searchResult) as TFile
-			const fileContentsOrEmpty = await app?.vault.read(abstractFile)
+			const abstractFile = this.app.vault.getAbstractFileByPath(searchResult)
```
Comment (kristenbrann) [inline src/main.ts:146]: don't access global `app`, instead using `this.app`

File: src/main.ts [hunk 4]
```diff
@@ -143,8 +143,12 @@ export default class VaultChat extends Plugin {
...
+			if (!(abstractFile instanceof TFile)) {
```
Comment (kristenbrann) [inline src/main.ts:147]: check if result is actually a file and not a folder

File: src/useApp.ts
Comment (kristenbrann) [inline src/useApp.ts:1]: unused code from initially incorporating react

**Review by cpaika — APPROVED** [review]

**Pull request merged**